src/groups/special_euclidean.jl

```diff
@@ -692,3 +692,27 @@ end
 function project!(M::SpecialEuclideanInGeneralLinear, Y, p, X)
     return copyto!(Y, project(M, p, X))
 end
+
+### Special methods for better performance of selected operations
+
+function exp(M::SpecialEuclidean, p::ArrayPartition, X::ArrayPartition)
```

Comment:
would it maybe make sense to change this to the right allocation methods (for the result) and implementing `exp!` instead?

Reply:
That's not really an option until Julia gets some form of inter-procedural allocation elision. Right now we can't mutate anything allocated on stack, and stack allocation is what we want here.

Reply:
Hm, this leave my German-ness for unified code a bit unsatisfied ;)

Reply:
Ah, but my other-german-ness has to add: Please document these 🤓

Reply:
Yes, it's not particularly satisfying and I hope Julia will get these optimizations.

How much documentation do you expect here? These methods don't introduce any new functionality.

Reply:
It would increase visibility? For now the user has to know that these exist to use them, right? Or are points on SE(n) already always ArrayPartitions anyways?

Reply:
I don't really intend to write a long explanation on performance for SE(n) here. These methods will be automatically used if user uses `ArrayPartition` with `SArray`, which is standard generic performance advice. Points on SE(n) aren't necessarily `ArrayPartition` but we can add similar optimizations for affine matrices too if needed.

Reply:
Ok.

Reply:
Should we maybe – for code clarity and structure – at some point aim for RecursiveArrayTools and StaticArrays becoming weak dependencies? That would collect these performance dispatches in the extension – so in one place?

Reply:
Yes, we could convert both libraries to weak dependencies. `RecursiveArrayTools` more easily than `StaticArrays` because the latter is used in some methods not restricted to StaticArrays types.

Reply:
Nice. I will open an issue later today to keep track of this (but we also have enough open PRs already by now ;))